docs/rfcs/1020-pki-support.md

```diff
@@ -0,0 +1,281 @@
+
+    > [!IMPORTANT]
+    > The certificate needs to have a `CommonName` ([oid-2.5.4.3](https://oid-base.com/get/2.5.4.3)) and a `EmailAddress` ([oid-1.2.840.113549.1.9.1](https://oid-base.com/get/1.2.840.113549.1.9.1)) attributes in the certificate's subject.
+
+ 4. Alice creates a user encryption key pair and a device signing key pair.
+ 5. Alice saves the private parts of the user & device keys pairs on the filesystem ([`pki_enrollment_create_local_pending`]). Those data are stored encrypted by the Smartcard's encryption key.
+ 6. Alice submits the request (authorized to re-submit later on with updated information) with: ([`pki_enrollment_submit`]):
+
+    - The PKI certificate.
+    - A payload consisting of the public parts of the user & device keys pairs and the desired device label ([`PkiEnrollmentSubmitPayload`]) (signed using [`pki_enrollment_sign_payload`]).
+
+ 7. Mallory (a CoolOrg admin which also has the PKI configured, this time the OS certificate store hold the private key and x509 certificate) sees the enrollment request from Alice ([`pki_enrollment_list`]).
+ 8. Mallory verifies the provided x509 certificate ([`pki_enrollment_load_peer_certificate`]) and provided data ([`pki_enrollment_load_submit_payload`]).
+
+    > [!NOTE]
+    > The verification of the certificate implies that we have access to the root certificates to be able to verify the trust chain.
+
+ 9. If Mallory accepts the request (after being validated), it creates the Alice's user and device certificates with the provided user & device public keys.
+10. Mallory signs both certificates with its device signing key. Then prepares the "accept" response payload ([`PkiEnrollmentAnswerPayload`])
+11. Mallory selects the x509 certificate (from the certificate store of the OS) ([`pki_enrollment_select_certificate`])
+12. Mallory sign the response payload with the certificate ([`pki_enrollment_sign_payload`])
+13. Mallory sends the signed accept response payload alongside the used x509 certificate to the server ([`pki_enrollment_accept`])
+
+    > [!NOTE]
+    > The payload destined to Alice is signed with the x509 certificate, but the request needs to be authenticated to the server so it's sign with Mallory's device.
+
+14. Alice checks the server to see the state of the request ([`pki_enrollment_info`])
+15. Alice retrieves the accept response from the server
+16. Alice starts validating the provided admin's x509 certificate ([`pki_enrollment_load_peer_certificate`]) and accept response payload signature ([`pki_enrollment_load_accept_payload`])
+17. From that accept response payload and the saved local pending part ([`pki_enrollment_load_local_pending_secret_part`]), Alice is able to create a local device and use it to access the organization.
+
+Alternative, if Mallory wasn't able to verify the payload (because on invalid certificate or signature) during step 8 or decided to reject the request at step 9,
```

Comment:
```suggestion
Alternative, if Mallory wasn't able to verify the payload (because of invalid certificate or signature) during step 8 or decided to reject the request at step 9,
```